README.md

```diff
@@ -189,6 +189,29 @@ Fahrschein supports different exponential backoff strategies when streaming even
 * `EqualJitterBackoffStrategy` (default) - extends `ExponentialBackoffStrategy` with the same defaults. For each delay it takes half of the delay value and adds the other half multiplied by a random factor [0..1).
 * `FullJitterBackoffStrategy` - extends `ExponentialBackoffStrategy` with the same defaults and multiplies each delay by a random factor [0..1).
 
+## Retries and exception handling for event publishing
+
+Fahrschein does not have sophisticated mechanisms for retry handling when publishing to Nakadi yet. The recommended way
+to handle exceptions when publishing is to create a retry-wrapper around the `NakadiClient.publish` method.
+
+In case of a partial success or also in cases like validation errors, which are complete failures, Fahrschein
+will throw an `EventPublishingException` with the `BatchItemResponse`s (as returned from Nakadi) for the failed
+ items in the responses property.
+
+These objects have the eid of the failed event, a `publishingStatus` (failed/aborted/submitted - but successful itemes are
+filtered out), the step where it failed and a detail string.
+
+If the application sets the eids itself (i.e. doesn't let Nakadi do it) and keeps track of them, this allows it
```

Comment:
Nakadi does not assign `eid` to events — the id has to be provided by the event publisher.

Reply:
https://github.com/zalando-nakadi/fahrschein/pull/422

Reply:
Huh, that's new to me. From my experience, when no `eid` is in the `metadata`, Nakadi will provide a random UUID. Did this change?

Reply:
occurred_at and eid are required fields. Nakadi enriches only those https://github.com/zalando/nakadi/blob/026476faf60e8294d206f014b4c667913fb34106/core-common/src/main/java/org/zalando/nakadi/enrichment/MetadataEnrichmentStrategy.java#L37-L42

Reply:
I just tried it, and events without `eid` are rejected. I guess I mixed this up with some other metadata fields. Sorry.